src/app.ts

```diff
@@ -13,14 +13,49 @@ interface TemperatureSummary {
   average: number
 }
 
+
+const redingMap: { [key: string]: { [key: string]: number[] } } = {}; 
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((reding) => {
+
+    const date = reding.time.toString().split('T')[0];
+    if (!redingMap[date]) {
+      redingMap[date] = {};
+    }
+
+    if (!redingMap[date][reding.city]) {
+      redingMap[date][reding.city] = [];
+    }
+
+    redingMap[date][reding.city].push(reding.temperature);
+
+  });
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
+
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const dateString = date.toString().split('T')[0];
+  
+  if (!redingMap[dateString] || !redingMap[dateString][city]) {
+    return null;
+  }
+
+  const temperatures = redingMap[dateString][city];
+  const first = temperatures[0];
+  const last = temperatures[temperatures.length - 1];
+  const high = Math.max(...temperatures);
+  const low = Math.min(...temperatures);
+  const average = temperatures.reduce((sum, temp) => sum + temp, 0) / temperatures.length;
+
+  return {
```

Comment:
you can also return the values directly right?

Reply:
I mean 
return {
  first: temperatures[0]
  ...

Reply:
Yes, I could have returned them directly, considering that I was not going to need them for other types of operations, thanks for the suggestion!